src/book.cpp

```diff
@@ -215,15 +212,42 @@ void Book::setPath(const std::string& path)
    : path;
 }
 
-const std::string& Book::getFavicon() const {
-  if (m_favicon.empty() && !m_faviconUrl.empty()) {
+const Book::Illustration Book::missingDefaultIllustration;
+
+const Book::Illustration& Book::getDefaultIllustration() const
+{
+  for ( const auto& ilPtr : m_illustrations ) {
+    if (ilPtr->width == 48 && ilPtr->height == 48) {
+      return *ilPtr;
+    }
+  }
+  return missingDefaultIllustration;
+}
+
+const std::string& Book::Illustration::getData() const
+{
+  if (data.empty() && !url.empty()) {
     try {
-      m_favicon = download(m_faviconUrl);
+      data = download(url);
     } catch(...) {
-      std::cerr << "Cannot download favicon from " << m_faviconUrl;
+      std::cerr << "Cannot download favicon from " << url;
     }
   }
-  return m_favicon;
+  return data;
```

Comment:
We probably want to protect this from race condition. Two "simultaneous" `Illustration::getData` calls may lead to two downloads.

Reply:
Should I use a per-illustration mutex, or a global shared mutex is fine?

Reply:
I went with a shared mutex.

Reply:
Another issue with `Book::Illustration::getData()` is that it will never stop trying to download the data if the URL is wrong or inaccessible. Shouldn't we set a limit on the count of download attempts or a minimum time interval between such attempts?

Reply:
A per-illustration mutex is better. If user code does multi-threading we probably don't want to introduce a single sync point, where we do only one download at the time.

Reply:
Fixed in an amended commit.